PWGLF/Tasks/QC/straRecoStudy.cxx

```diff
@@ -79,27 +79,80 @@ struct preProcessMCcollisions {
 
   HistogramRegistry histos{"Histos", {}, OutputObjHandlingPolicy::AnalysisObject};
 
+  Preslice<aod::Tracks> perCollision = aod::track::collisionId;
+
+  struct collisionStats { // This structure is named "myDataType"
+    float nContribsWithTPC;
+    float nContribsWithTRD;
+    float nContribsWithTOF;
+    float nContribsWithITS;
+  };
+
+  template <typename T>
+  std::vector<std::size_t> sort_indices(const std::vector<T>& v)
+  {
+    std::vector<std::size_t> idx(v.size());
+    std::iota(idx.begin(), idx.end(), 0);
+    std::stable_sort(idx.begin(), idx.end(),
+                     [&v](std::size_t i1, std::size_t i2) { return v[i1] > v[i2]; });
+    return idx;
+  }
+
   void init(InitContext const&)
   {
     const AxisSpec axisNTimesCollRecoed{(int)10, -0.5f, +9.5f, ""};
+    const AxisSpec axisTrackCount{(int)50, -0.5f, +49.5f, ""};
     histos.add("NTimesCollRecoed", "NTimesCollRecoed", kTH1F, {axisNTimesCollRecoed});
+
+    // A trick to store more information, please
+    histos.add("h2dTrackCounter", "h2dTrackCounter", kTH2D, {axisTrackCount, axisNTimesCollRecoed});
   }
 
-  void process(aod::McCollisions const& mccollisions, soa::Join<aod::Collisions, aod::McCollisionLabels> const& collisions)
+  void process(aod::McCollision const& mcCollision, soa::SmallGroups<soa::Join<aod::McCollisionLabels, aod::Collisions>> const& collisions, TracksCompleteIUMC const& tracks)
   {
-    for (auto& mccollision : mccollisions) {
-      // check if collision successfully reconstructed
-      int lExists = 0;
-      for (auto& collision : collisions) {
-        if (collision.has_mcCollision()) {
-          if (mccollision.globalIndex() == collision.mcCollision().globalIndex()) {
-            lExists++;
-          }
+    std::vector<collisionStats> collisionStats;
+    std::vector<int> collisionNContribs;
+    collisionStats.reserve(collisions.size());
+    // collisionNContribs.reserve(collisions.size());
+    for (Int_t ic = 0; ic < collisions.size(); ic++) {
```

Comment:
It would be easier to use the default zeros in the struct definition and instead of reserving the vector you can fill it with N default-constructed structs on construction.

Reply:
Ah sure, this I can do, fair enough :-)